plasmoid/contents/ui/screens.js

```diff
@@ -0,0 +1,49 @@
+function saveConfig() {
+    console.log("Saving config")
+    cfg_screenNameList = [];
+    cfg_screenColorList = [];
+
+    for (var i = 0; i < screenListModel.count; i++) {
+        var name = screenListModel.get(i).screenName;
+        var color = Qt.lighter(screenListModel.get(i).screenColor, 1);
+        color.a = cfg_screenColorOpacity;
+
+        cfg_screenNameList.push(name);
+        cfg_screenColorList.push(color);
+    }
+}
+
+function loadConfig() {
+    for (var i = 0; i < cfg_screenNameList.length; i++) {
+        var name = cfg_screenNameList[i];
+        var color = Qt.lighter(cfg_screenColorList[i], 1);
+        color.a = 1;
+
+        screenListModel.append({screenName: name, screenColor: color});
+    }
+}
+
+function loadNewScreens() {
+    var defaultColors = ["#004040", "#007D7D", "#00D7D7"];
+    for (var screenName of i3pager.getScreenNames()) {
+        if (!isScreenNameOnList(screenName, cfg_screenNameList)) {
+            if (screenListModel.count >= defaultColors.length) {
+                screenListModel.append({screenName: screenName, screenColor: defaultColors[2]});
+            } else {
+                screenListModel.append({
+                    screenName: screenName,
+                    screenColor: defaultColors[screenListModel.count]
+                });
```

Comment:
Since the only difference here is the `defaultColors` index I think it's easier to read if the index is first selected and then the append happens afterwards. Could be done by doing something like `defaultColorIndex = Math.min(screenListModel.count, defaultColors.length - 1)`.

Reply:
```js
function loadNewScreens() {
    var defaultColors = ["#004040", "#007D7D", "#00D7D7"];
    for (var screenName of i3pager.getScreenNames()) {
        if (!isScreenNameOnList(screenName, cfg_screenNameList)) {
            screenListModel.append({
                screenName: screenName,
                screenColor: defaultColors[Math.min(screenListModel.count, defaultColors.length - 1)]
            });
        }
    }
}
```
Something like this? Do I understand correctly?

Reply:
Yeah, however I would create a variable for the index to reduce the amount of things happening on one line. That's a personal preference though.
```js
        if (!isScreenNameOnList(screenName, cfg_screenNameList)) {
            var colorIndex = Math.min(screenListModel.count, defaultColors.length - 1);
            screenListModel.append({
                screenName: screenName,
                screenColor: defaultColors[colorIndex]
            });
        }
```